utils/data_version_control.py

```diff
@@ -98,3 +102,63 @@ def version_new_data(filename_roots: List[str]) -> None:
     dvc_files = [f"{fname}.csv.dvc" for fname in filename_roots]
     add_csv_files_to_dvc(csv_files)
     add_and_commit_dvc_files_to_git(dvc_files)
+
+
+def git_checkout_folder(
+    tag_name: Optional[str] = None,
+    commit_hash: Optional[str] = None,
+    folder_name: str = "data",
+) -> None:
+    """Checkout a specified folder within a tagged commit or commit hash on Git.
+
+    Args:
+        tag_name (Optional[str]): Git tag to checkout.
+        commit_hash (Optional[str]): Git commit hash to checkout.
+        folder_name (str): Folder name to checkout e.g. 'data'.
+    """
+    if tag_name is not None and git_tag_exists(tag_name):
+        g = git.cmd.Git(PROJECT_ROOT_DIR)
+        g.checkout(tag_name, folder_name)
+    elif commit_hash is not None and git_commit_hash_exists(commit_hash):
+        g = git.cmd.Git(PROJECT_ROOT_DIR)
+        g.checkout(commit_hash, folder_name)
+    else:
+        logger.error("Error, you must specify a valid tag or commit hash.")
+
+
+def git_tag_exists(tag_name: str) -> bool:
+    """Checks for the existance of a given Git tag name.
+
+    Args:
+        tag_name (str): Name of tag to check existence.
+
+    Returns:
+        bool: True, if Git tag exists. False otherwise.
+    """
+    try:
```

Comment:
This (and the function below) could be simplified: `return tag_name in git.Repo().tags.keys()`